### PR TITLE
Allow halo zones around FD elements

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/DgSubcell/SubcellOptions.hpp [hunk 1]
```diff
@@ -102,11 +102,28 @@ class SubcellOptions {
         "solution."};
     using type = fd::ReconstructionMethod;
   };
+  /// \brief Use a one-wide halo of FD elements around any troubled element.
+  ///
+  /// This provide a buffer of FD subcells so that as a discontinuity moves
```
Comment (kidder) [inline src/Evolution/DgSubcell/SubcellOptions.hpp]: provides

File: src/Evolution/DgSubcell/SubcellOptions.hpp [hunk 2]
```diff
@@ -102,11 +102,28 @@ class SubcellOptions {
         "solution."};
     using type = fd::ReconstructionMethod;
   };
+  /// \brief Use a one-wide halo of FD elements around any troubled element.
```
Comment (kidder) [inline src/Evolution/DgSubcell/SubcellOptions.hpp]: one-wide -> width-one

File: src/Evolution/DgSubcell/SubcellOptions.hpp [hunk 3]
```diff
@@ -102,11 +102,28 @@ class SubcellOptions {
...
+  /// This provide a buffer of FD subcells so that as a discontinuity moves
+  /// from one element to another we do not get any Gibbs phenomenon. In the
+  /// case where we evolve the spacetime metric (e.g. GH+GRMHD) a halo region
+  /// provides a buffer in case the stellar surface is near an element
+  /// boundary. Since the GH variables are interpolated using high-order
+  /// unlimited reconstruction, they can run into issues with Gibbs phenomenon.
+  struct UseHalo {
+    using type = bool;
+    static constexpr Options::String help = {
+        "Use a one-wide halo of FD elements around any troubled element."
```
Comment (kidder) [inline src/Evolution/DgSubcell/SubcellOptions.hpp]: one-wide -> width-one

File: src/Evolution/DgSubcell/SubcellOptions.hpp [hunk 4]
```diff
@@ -102,11 +102,28 @@ class SubcellOptions {
...
+        "\n"
+        "This provide a buffer of FD subcells so that as a discontinuity "
```
Comment (kidder) [inline src/Evolution/DgSubcell/SubcellOptions.hpp]: provides

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash

**Comment by nilsdeppe** [comment]
Rebased and squashed :) Thanks for the review!

**Review by kidder — APPROVED** [review]

**Comment by nilsdeppe** [comment]
Will merge as part of #4620 That will help improve CI flow.

**Pull request closed without merging**